prow/test/integration/test/webhook_server_test.go

```diff
@@ -118,8 +126,151 @@ func TestWebhookServerMutateProwjob(t *testing.T) {
 	}
 }
+
+func TestWebhookServerAvailability(t *testing.T) {
+	
+	clusterContext := getClusterContext()
+	kubeClient, err := NewClients("", clusterContext)
+	if err != nil {
+		t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
+	}
+	ctx := context.Background()
+	t.Cleanup(func() {
+		var testProwjob v1.ProwJob
+		absDeploymentPath, err := filepath.Abs(deploymentPath)
+		if err != nil {
+			logrus.WithError(err).Info("could not find absolute path")
+		}
+		out, err := exec.Command("kubectl", "--context="+getClusterContext(), "apply", "-f", absDeploymentPath).CombinedOutput()
+		if err != nil {
+			logrus.WithError(err).Infof("could not clean up cluster %s", string(out))
+		}
+		key := types.NamespacedName{
+			Namespace: "default",
+			Name: testServerAvailability,
+		}
+		if err := kubeClient.Get(ctx, key, &testProwjob); err != nil {
+			logrus.WithError(err).Infof("prowjob not found")
+		} else {
+			out, err := exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", testServerAvailability).CombinedOutput()
+			if err != nil {
+				logrus.WithError(err).Infof("could not clean up cluster %s", string(out))
+			}	
+		}
+		
+	})
+	
+	key := types.NamespacedName{
+		Namespace: "default",
+		Name:      "webhook-server",
+	}
+	var found bool
+	var webhookDeployment appsv1.Deployment
+	err = kubeClient.Get(ctx, key, &webhookDeployment)
+	if err != nil {
+		t.Fatalf("could not list deployment %v", err)
+	}
+	found = true
+	originalProwJob := &v1.ProwJob{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ProwJob",
+			APIVersion: "prow.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"foo":               "bar",
+				"default-me":        "enabled",
+				"admission-webhook": "enabled",
+			},
+			Name:      testServerAvailability,
+			Namespace: "default",
+		},
+		Spec: v1.ProwJobSpec{
+			Cluster: "foo",
+			Refs: &v1.Refs{
+				Repo: "bar",
+			},
+			Agent:  v1.TektonAgent,
+			Type:   v1.PresubmitJob,
+			Report: true,
+		},
+		Status: v1.ProwJobStatus{
+			State: v1.PendingState,
+		},
+	}
+	webhookDeploymentCopy := webhookDeployment.DeepCopy()
+	webhookDeploymentCopy.Spec.Template.Spec.Containers[0].LivenessProbe.PeriodSeconds = 4
+	var podList corev1.PodList
+	var webhookServerPod corev1.Pod
+	err = kubeClient.List(ctx, &podList, &client.ListOptions{})
+	if err != nil {
+		t.Fatalf("could not list pods %v", err)
+	}
+	for _, pod := range podList.Items {
+		if pod.ObjectMeta.Labels["app"] == "webhook-server" {
+			webhookServerPod = pod
+			found = true
+			break
+		}
+	}
+	var wg sync.WaitGroup
+	if !found {
+		t.Fatalf("webhook-server is not running")
+	} else {
+		wg.Add(1)
+		oldName := webhookServerPod.GetName()
+		go func() {
+			defer wg.Done()
+			for {
+				exist, newPod, err := checkNewPodInCluster(oldName, kubeClient, ctx)
+				if err != nil {
+					return
+				}
+				count := 0
+				if exist && newPod.Status.Phase == "Running" {
+					return
```

Comment:
There is a chance that the oldPod is still running when the newPod is up, so I think the end goal should be oldPod is completely deleted and the newPod is running.

Also I think we should do one more round to mutating prowjob here to make sure that it's the newPod working.

Reply:
done the extra round of mutating checking the old pod is deleted has some unexpected behavior